MapboxCoreNavigation/RouterDelegate.swift

```diff
@@ -4,43 +4,156 @@ import MapboxDirections
 
 
 
-//:nodoc: See NavigationServiceDelegate.
+/**
+ RouterDelegate is a mid-level delegate that defines the messaging architechture between the NavigationService and its router. Developers generally shouldn't have to concern themselves with this class unless a custom core-navigation engine (conforming to NavigationService) is being written.
+ 
+ You can use this protocol to:
```

Comment:
This list contradicts the paragraph above it: either developers should ignore this class or they should consider it for all the things below. Here’s a way to phrase it that also avoids discussing the SDK’s architecture, which shouldn’t be the concern of public, class-level documentation:

> A router delegate interacts with one or more `Router` instances, such as `RouteController` objects, during turn-by-turn navigation. This protocol is similar to `NavigationServiceDelegate`, and the main conforming class is `MapboxNavigationService`. Normally you should not need to make a class conform to the `RouterDelegate` protocol or call any of its methods directly, but you would need to call this protocol’s methods if you implement a custom `Router` class.
> 
> As the user progresses along a route, a router informs its delegate about significant events as they occur, and the delegate has opportunities to influence the route and its presentation. For example, when the navigation service reports that the user has arrived at the destination, your delegate implementation could present information about the destination. It could also customize individual visual or spoken instructions along the route by returning modified instruction objects.